overlord/state/taskrunner.go

```diff
@@ -46,6 +46,17 @@ func (r *Retry) Error() string {
 	return "task should be retried"
 }
 
+// Hold is returned from a handler to signal that the task cannot
+// proceed at the moment maybe because some manual action from the
+// user required at this point or because of errors.
+type Hold struct {
```

Comment:
nitpick: I see that this follows `Retry`'s pattern in that it's an error but the taskrunner handles it in a special way. The name doesn't make it clear that is an error but maybe that's intended since it's not used a typical one

Reply:
yes, that's intentional that way